.buildkite/build_pr_pipeline.yml

```diff
@@ -1,7 +1,18 @@
 steps:
-  - label: ":white_check_mark: Build docs PR"
-    command: |
-      bash .buildkite/scripts/build_pr.sh
+  - key: "build-pr-setup"
+    label: "setup"
+    command: ".buildkite/scripts/build_pr_setup.sh"
+  - key: "build-pr"
+    label: ":hammer: Build docs PR"
+    command: ".buildkite/scripts/build_pr.sh"
+    depends_on:
+      - step: build-pr-setup
+        allow_failure: true
```

Comment:
I was reading a bit about how Buildkite [step dependencies](https://buildkite.com/docs/pipelines/dependencies) work. Do we want to continue building the PR if the setup step fails? It seems like the script's primary purpose is to update the PR status checks to show this is in progress. 

It deliberately exits if not run by the Buildkite PR bot. It might be helpful to stop if that fails for whatever reason (but if we don't put a "pending" status check, is that really the worst thing?

Reply:
I agree that missing the pending state is not the end of the world, and worse - the person opening the PR may not see that the pending state was not added, and assuming the build was started - so we may as well start it.

A case could be made to fail the build for the passing/failing state at the end.